lmdeploy/serve/openai/protocol.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+# Modified from
+# https://github.com/lm-sys/FastChat/blob/168ccc29d3f7edc50823016105c024fe2282732a/fastchat/protocol/openai_api_protocol.py
+import time
+from typing import Any, Dict, List, Literal, Optional, Union
+
+import shortuuid
+from pydantic import BaseModel, Field
+
+
+class ErrorResponse(BaseModel):
+    """Error responses."""
+    object: str = 'error'
+    message: str
+    code: int
+
+
+class ModelPermission(BaseModel):
+    """Model permissions."""
+    id: str = Field(default_factory=lambda: f'modelperm-{shortuuid.random()}')
+    object: str = 'model_permission'
+    created: int = Field(default_factory=lambda: int(time.time()))
+    allow_create_engine: bool = False
+    allow_sampling: bool = True
+    allow_logprobs: bool = True
```

Comment:
Actually, turbomind doesn't support `logprobs`

Reply:
Yes, note this in the openai api docstring